common/lib/xmodule/xmodule/js/spec/video/sjson_spec.js

```diff
@@ -30,6 +30,31 @@ function (Sjson) {
             expect(sjson.search(21620)).toEqual(4);
             expect(sjson.search(24920)).toEqual(5);
         });
+
+        it('search returns -1 for a value outside the bounds of the array', function() {
+            expect(sjson.search(10024920)).toEqual(sjson.getCaptions().length - 1);
+        });
+
+        it('search only searches through a subrange of times if start / end times are specified', function () {
+            var results = sjson.filter(6000, 10000);
+            var expectedLength = results.captions.length - 1;
+
+            expect(sjson.search(0, 6000, 10000)).toEqual(-1);
+            expect(sjson.search(3120, 6000, 10000)).toEqual(-1);
+            expect(sjson.search(6270, 6000, 10000)).toEqual(0);
+            expect(sjson.search(8490, 6000, 10000)).toEqual(0);
+            expect(sjson.search(10021620, 6000, 10000)).toEqual(expectedLength);
+        });
+
+        it('filters results correctly given a start and end time', function () {
+            var results = sjson.filter(3100, 20000); 
+
+            expect(results).toBeDefined();
+            expect(results.start).toBeDefined();
```

Comment:
I think we can actually do without this check; if `results.start` is not defined, the following checks on `results.start.length` will fail.